lib/index.coffee

```diff
@@ -14,8 +14,14 @@ class DockerGC
 		@hostObj = _.defaults({ Promise }, hostObj)
 		@dockerProgress = new DockerProgress(@hostObj)
 		dockerUtils.getDocker(@hostObj)
-		.then (docker) =>
-			@docker = docker
+		.then (@docker) =>
+			# Docker info can take a while so do it here,
+			# and don't wait on the results
+			@baseImagePromise = @getDaemonArchitecture()
+			.then (arch) ->
+				return switch arch
```

Comment:
There should probably be an else if, if for nothing else then to at least add a warning that something weird has happened

Reply:
I've thrown an error, we don't really want to continue in that case.